src/components/templates/Camp/components/Map.tsx

```diff
@@ -42,22 +52,43 @@ export const Map: React.FC<PropsType> = ({
               width: room.width_percent + "%",
               height: room.height_percent + "%",
             }}
-            key={idx}
-            onClick={() => openRoomModal(room)}
+            key={room.title}
+            onClick={() => {
+              setClicked((prev) => !prev);
+              setRoomClicked(room.title);
+            }}
           >
-            <img
-              className="room-image"
-              src={room.image_url}
-              onClick={() => openRoomModal(room)}
-              title={room.title}
-              alt={room.title}
-            />
-            <RoomAttendance
-              roomTitle={room.title}
-              attendance={attendances[room.title]}
-              key={idx}
-              onClick={() => openRoomModal(room)}
-            />
+            <div className={`playa-venue ${clicked ? "clicked" : ""}`}>
+              <div className="playa-venue-img">
+                <img src={room.image_url} title={room.title} alt={room.title} />
+              </div>
+              <div className="playa-venue-text">
+                <div className="playa-venue-maininfo">
+                  <div className="playa-venue-title">{room.title}</div>
+                  <div className="playa-venue-people">
+                    {attendances[room.title] || 0}
+                  </div>
+                </div>
+                <div className="playa-venue-secondinfo">
+                  <div className="playa-venue-desc">
+                    <p>{room.subtitle}</p>
+                    <p>{room.about}</p>
+                  </div>
+                  <div className="playa-venue-actions">
+                    <a
+                      className="btn btn-block btn-small btn-primary"
+                      href={
+                        room.url.includes("http") ? room.url : "//" + room.url
```

Comment:
not a fan of this sort of logic being hidden inside the JSX. Move into a function outside of component